### PR TITLE
En/merchant controller

### PR DESCRIPTION
**Pull request by north108**
Works Controller about 80% of the way there. Tests are done and passing for what I wrote.

### CODE REVIEW

File: test/controllers/merchants_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  #let(:merchant) { merchants(:brad) }
+
+  describe "index" do
+    it "responds with success when there are many merchants" do
+      get merchants_path
+      must_respond_with :success
+    end
+  
+  
+    it "responds with success when there are no merchants" do
```
Comment (TiffanyChio) [inline test/controllers/merchants_controller_test.rb:15]: If there are merchants in your YAML, you'll need to delete all of them first within your "it" block before the rest of the logic

File: test/controllers/merchants_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    end
+
+    it "redirects to the login if given invalid merchant data" do 
+      start_count = Merchant.count
+
+      merchant = merchants(:brad)
+      merchant.uid = nil
+      merchant.save
+      p merchant
```
Comment (TiffanyChio) [inline test/controllers/merchants_controller_test.rb:71]: Might need to delete p merchant?

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :merchants, except: [:delete]
+  get "/auth/github", as: "github_login"
```
Comment (krismosk) [inline config/routes.rb:4]: Consider adding the route: get "/users/current", to: "users#current", as: "current_user"

File: app/controllers/merchants_controller.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      merchant = Merchant.build_from_github(auth_hash)
+
+      if merchant.save
+        flash[:success] = "Logged in as new merchant #{merchant.username}."
+      else
+        flash[:error] = "Could not create new merchant account: #{merchant.errors.messages}"
+        return redirect_to root_path
+      end 
+    end
+
```
Comment (krismosk) [inline app/controllers/merchants_controller.rb:31]: Consider adding the current_user controller action to track the presence of the current session user id.

Comment (krismosk) [reply app/controllers/merchants_controller.rb:31]: @current_user = User.find_by(id: session[:user_id])
    unless @current_user
      flash[:error] = "You must be logged in to see this page"
      redirect_to root_path
    end
  end

**Review by TiffanyChio — APPROVED** [review]
Approved. Some minor changes requested in controller test file that can be done at later date.

**Review by krismosk — APPROVED** [review]

**Pull request merged**